### DISCUSSION TITLE
How to allow case statements in AllowedPatterns or AllowedMethods

### DISCUSSION THREAD

**Discussion by aaronmallen** [rubocop/rubocop#13651] [Q&A] [answered]
RuboCop introduced the AllowedPatterns, AllowedMethods, and CountAsOne configuration options for the Metrics/MethodLength cop. However, the documentation does not provide clear examples of how to use AllowedPatterns and AllowedMethods.

Specifically, I am looking for guidance on how to configure AllowedPatterns to allow methods containing case statements to be ignored by the Metrics/MethodLength cop. Any examples or clarifications on how to achieve this would be greatly appreciated.

> Note: this may be related to rubocop/ruby-style-guide#702

**Comment by dvandersluis** [comment, marked as answer]
For `Metrics/MethodLength`, `AllowedMethods` and `AllowedPatterns` are only considering the method name. `AllowedMethods` takes an array of discrete names, and `AllowedPatterns` takes an array of regular expressions. In both cases, they are matched **only** against the method name to determine if the method should be allowed when exceeding the configured maximum size.

`CountAsOne` is used to tell `Metrics/MethodLength` to consider some multi-line constructs as taking a single line. These are `array`, `hash`, `heredoc` and `method_call`.  You can see examples of these four configurations [here](https://docs.rubocop.org/rubocop/cops_metrics.html#countasone_-__array__-_hash__-_heredoc__-_method_call__-metricsmethodlength).

There isn't a way to allow methods based on content. The best way to do this would be to wrap your methods you want to exclude with `# rubocop:disable`...`# rubocop:enable` comments.
